### ISSUE TITLE
Error with util install-vpc-controllers when trying to create windows nodes.

### ISSUE THREAD

**Issue by dashington** [weaveworks/eksctl#1452]
**What happened?**
 I tried to use eksctl to create a windows EKS cluster.  I was able to successfully create a linux cluster following the steps in https://eksworkshop.com/ using eskctl from cloud9.

But when trying to set up a windows cluster using the same environment and following the steps in the the blog https://aws.amazon.com/blogs/aws/amazon-eks-windows-container-support-now-generally-available/ and the documentation https://docs.aws.amazon.com/eks/latest/userguide/windows-support.html

it fails on the step...
*eksctl utils install-vpc-controllers --name demo-windows-cluster --approve*
  with the error
_[:information_source:]  using region us-west-2_
_[:information_source:]  replaced "ClusterRole.rbac.authorization.k8s.io/vpc-resource-controller"_
_[:information_source:]  replaced "ClusterRoleBinding.rbac.authorization.k8s.io/vpc-resource-controller"_
_[:information_source:]  replaced "kube-system:ServiceAccount/vpc-resource-controller"_
_[:information_source:]  replaced "kube-system:Deployment.apps/vpc-resource-controller"_
_[:information_source:]  created "CertificateSigningRequest.certificates.k8s.io/vpc-admission-webhook.kube-system"_
[:heavy_multiplication_x:]  _error installing VPC controller: failed to find certificate after approval_

If I run that command a second time, the error goes away ...
_[:information_source:]  using region us-west-2_
_[:information_source:]  replaced "ClusterRole.rbac.authorization.k8s.io/vpc-resource-controller"_
_[:information_source:]  replaced "ClusterRoleBinding.rbac.authorization.k8s.io/vpc-resource-controller"_
_[:information_source:]  replaced "kube-system:ServiceAccount/vpc-resource-controller"_
_[:information_source:]  replaced "kube-system:Deployment.apps/vpc-resource-controller"_
_[:information_source:]  replaced "kube-system:Service/vpc-admission-webhook"_
_[:information_source:]  replaced "kube-system:Deployment.apps/vpc-admission-webhook"_
_[:information_source:]  replaced "kube-system:MutatingWebhookConfiguration.admissionregistration.k8s.io/vpc-admission-webhook-cfg"_

and I was able to add windows nodes.  However the sample app, windows-server-IIS.yaml, containers never start up so clearly there is a problem.  

Since I was using a corporate account with some limits on permissions (IAM and Organization) are specific permissions I know I don't have,  I created a personal AWS account and tried the same thing again...successfully following the workshop using eskctl and creating the cluster, then removing it and attempting to create the windows cluster as per the blog.  I got the same error running...
*eksctl utils install-vpc-controllers --name demo-windows-cluster --approve*
...
[:heavy_multiplication_x:]  _error installing VPC controller: failed to find certificate after approval_

I also tried this command on the cluster created according to the workshop instructions with the same result. 

I  attempting to create the sample application in my personal account after seeing the error with install-vpc-controllers and re-running it to hide the error as above with the same result.

I have not tried the above outside Cloud9 but previously I have only managed to get EKS to work using eskctl in Cloud9.  I was never able to get the non eskctl method to work to create a cluster I guessed due to IAM authorization problems.  I found the documentation is vague complex and assumes too much.

**What you expected to happen?**
I expect to be able to create a Windows nodes using eskctl and be able to run the sample application.

**How to reproduce it?**
see above

**Anything else we need to know?**
Using Firefox running on Windows 10 to access AWS, Cloud9 etc but doubt that matters.
I originally posted the problem on the #eks slack group as Tim

**Versions**
Please paste in the output of these commands:
```
$ eksctl version
[ℹ]  version.Info{BuiltAt:"", GitCommit:"", GitTag:"0.7.0"}
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.7", GitCommit:"4683545293d792934a7a7e12f2cc47d20b2dd01b", GitTreeState:"clean", BuildDate:"2019-06-06T01:46:52Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.7-eks-e9b1d0", GitCommit:"e9b1d0551216e1e8ace5ee4ca50161df34325ec2", GitTreeState:"clean", BuildDate:"2019-09-21T08:33:01Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}```

**Logs**
see above

**Comment by kerogers-cloudbees** [comment]
Seeing the same error running the command from our CI environment. 
```23:37:11  + eksctl utils install-vpc-controllers --name ci --region us-east-1 --approve
23:37:11  [ℹ]  using region us-east-1
23:37:11  [ℹ]  created "ClusterRole.rbac.authorization.k8s.io/vpc-resource-controller"
23:37:11  [ℹ]  created "ClusterRoleBinding.rbac.authorization.k8s.io/vpc-resource-controller"
23:37:11  [ℹ]  created "kube-system:ServiceAccount/vpc-resource-controller"
23:37:11  [ℹ]  created "kube-system:Deployment.apps/vpc-resource-controller"
23:37:11  [ℹ]  created "CertificateSigningRequest.certificates.k8s.io/vpc-admission-webhook.kube-system"
23:37:11  [✖]  error installing VPC controller: failed to find certificate after approval```

Versions:
```23:22:42  + eksctl version
23:22:42  [ℹ]  version.Info{BuiltAt:"", GitCommit:"", GitTag:"0.7.0"}```

```23:37:11  + kubectl version
23:37:11  Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.6", GitCommit:"96fac5cd13a5dc064f7d9f4f23030a6aeface6cc", GitTreeState:"clean", BuildDate:"2019-08-19T11:13:49Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
23:37:11  Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.7-eks-e9b1d0", GitCommit:"e9b1d0551216e1e8ace5ee4ca50161df34325ec2", GitTreeState:"clean", BuildDate:"2019-09-21T08:33:01Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}```

**Comment by cPu1** [comment]
Thanks for the detailed report, I'll be looking into this issue.

>and I was able to add windows nodes. However the sample app, windows-server-IIS.yaml, containers never start up so clearly there is a problem.

Windows Server 2019 container images are quite bulky, so it may take some time to pull the image. You can confirm this by checking the events from the windows-iis pod (`kubectl describe pod`) and see if the last event is "pulling image" (the pod would be in `Waiting` state).  If you wait for the image pull to complete, your pod should be up.

If the VPC controller components aren't set up properly, you may see something like this:
```failed create pod sandbox: rpc error: code = Unknown desc = [failed to set up sandbox container "9ced5d0be1063b2443c81951ec82a779ed55d08ead8deb8bc9b0e8bd33ae42bb" network for pod "windows-server-iis-7dcfc7c79b-vlbzj"```

As a workaround until the issue is fixed, running `eksctl utils install-vpc-controllers --name demo-windows-cluster --approve` a second time (after waiting for a few seconds) should fix the issue.

**Comment by dashington** [comment]
I noted in the issue that running eksctl utils install-vpc-controllers --name demo-windows-cluster --approve a second time seemed to clear or hide the "certificate" error.  It shows up again if you run the command a 3rd time.

**Comment by kerogers-cloudbees** [comment]
Same situation, after running twice, the vpc-admission-webhook pod is not exiting ContainerCreating. 

These are in the pod events:
```  Warning  FailedMount  2m8s (x3 over 6m40s)  kubelet, ip-192-168-51-132.ec2.internal  Unable to mount volumes for pod "vpc-admission-webhook-58956459f8-42bzr_kube-system(af628e42-f126-11e9-91a8-02546ec3a54c)": timeout expired waiting for volumes to attach or mount for pod "kube-system"/"vpc-admission-webhook-58956459f8-42bzr". list of unmounted volumes=[webhook-certs]. list of unattached volumes=[webhook-certs default-token-dbkhp]
  Warning  FailedMount  29s (x12 over 8m43s)  kubelet, ip-192-168-51-132.ec2.internal  MountVolume.SetUp failed for volume "webhook-certs" : secret "vpc-admission-webhook-certs" not found```

**Comment by kerogers-cloudbees** [comment]
Also I just noticed that the vpc-resource-controller secret appears twice. Is one of them the webhook certs and it got misnamed?

kube-system          vpc-resource-controller-token-6thgw                     kubernetes.io/service-account-token   3      16m
kube-system          vpc-resource-controller-token-c5r5b                     kubernetes.io/service-account-token   3      16m

**Comment by cPu1** [comment]
Since the webhook started before the secret was created, deleting the `vpc-admission-webhook` pod should fix the issue by triggering the deployment's ReplicaSet to create a new pod which will be able to mount the secret.

I have a PR that should fix this issue: #1456

**Comment by kerogers-cloudbees** [comment]
I don't think the secret was created, there's no secret containing "vpc-admission-webhook-certs" in its name.

**Comment by cPu1** [comment]
Was your second invocation of `eksctl utils install-vpc-controllers ` successful? If it was, you should see the secret creation logged in the output. Also, these resources are created in `kube-system`, can you confirm this is the namespace you used for checking the secret? `kubectl get secret -nkube-system vpc-admission-webhook-certs`

**Comment by dashington** [comment]
after the second eksctl utils install-vpc-controllers --name demo-windows-cluster --approve which does not report an error I get...
$ kubectl get secret -nkube-system vpc-admission-webhook-certs
Error from server (NotFound): secrets "vpc-admission-webhook-certs" not found

**Comment by cPu1** [comment]
>eksctl utils install-vpc-controllers --name demo-windows-cluster --approve which does not report an error I get...

Could you please share the output from this command? If that command succeeds, the `vpc-admission-webhook-certs` secret should exist.

>I'm not sure how to try your fix without building eksctl myself.

Building eksctl from `master` should be straightforward if you have the Go toolchain installed. Or you can wait for this fix be released, which will happen this week.

**Comment by cPu1** [comment]
@dashington this has been fixed in the new release: https://github.com/weaveworks/eksctl/releases/tag/0.8.0-rc.0

Could you try it out and let us know if it resolves your issue?

**Comment by dashington** [comment]
Apologies for the delay in getting back to this...working on other things.

I tried **eksctl 0.8.0-rc.0** and it appears to have fixed the issue I raised and "eksctl utils install-vpc-controllers..."  now succeeds and the **vpc-admission-webhook** pod now runs .  For reference below is the output I now see.  The only obvious oddity is the "[!]  certificate not yet available (event: ADDED)" message which I presume means the certificates were  ADDED.  I did not run  "eksctl utils install-vpc-controllers..." a seond time.

**eksctl version**[ℹ]  version.Info{BuiltAt:"", GitCommit:"", GitTag:"0.8.0-rc.0"}

**eksctl utils install-vpc-controllers --name demo-windows-cluster --approve**
Flag --name has been deprecated, use --cluster
[ℹ]  using region us-west-2
[ℹ]  created "ClusterRole.rbac.authorization.k8s.io/vpc-resource-controller"
[ℹ]  created "ClusterRoleBinding.rbac.authorization.k8s.io/vpc-resource-controller"
[ℹ]  created "kube-system:ServiceAccount/vpc-resource-controller"
[ℹ]  created "kube-system:Deployment.apps/vpc-resource-controller"
[ℹ]  created "CertificateSigningRequest.certificates.k8s.io/vpc-admission-webhook.kube-system"
[ℹ]  waiting for certificate to be available
**[!]  certificate not yet available (event: ADDED)**
[ℹ]  created "kube-system:secret/vpc-admission-webhook-certs"
[ℹ]  created "kube-system:Service/vpc-admission-webhook"
[ℹ]  created "kube-system:Deployment.apps/vpc-admission-webhook"
[ℹ]  created "kube-system:MutatingWebhookConfiguration.admissionregistration.k8s.io/vpc-admission-webhook-cfg"

**eksctl create nodegroup --region us-west-2 --cluster demo-windows-cluster --version 1.14 --name windows-ng --node-type t3.medium --nodes 3 --nodes-min 1 --nodes-max 4 --node-ami-family WindowsServer2019FullContainer --node-ami ami-0acf2fb05e6133356**
[ℹ]  eksctl version 0.8.0-rc.0
[ℹ]  using region us-west-2
[ℹ]  1 nodegroup(s) that already exist (standard-workers) will be excluded
[ℹ]  nodegroup "windows-ng" will use "ami-0acf2fb05e6133356" [WindowsServer2019FullContainer/1.14]
[ℹ]  1 nodegroup (windows-ng) was included (based on the include/exclude rules)
[ℹ]  combined exclude rules: standard-workers
[ℹ]  no nodegroups present in the current set were excluded by the filter
[ℹ]  will create a CloudFormation stack for each of 1 nodegroups in cluster "demo-windows-cluster"
[ℹ]  1 task: { create nodegroup "windows-ng" }
[ℹ]  building nodegroup stack "eksctl-demo-windows-cluster-nodegroup-windows-ng"
[ℹ]  deploying stack "eksctl-demo-windows-cluster-nodegroup-windows-ng"
[ℹ]  adding identity "arn:aws:iam::719681605826:role/eksctl-demo-windows-cluster-nodeg-NodeInstanceRole-1HWITR4U9LX5R" to auth ConfigMap
[ℹ]  nodegroup "windows-ng" has 0 node(s)
[ℹ]  waiting for at least 1 node(s) to become ready in "windows-ng"
[ℹ]  nodegroup "windows-ng" has 2 node(s)
[ℹ]  node "ip-192-168-34-150.us-west-2.compute.internal" is ready
[ℹ]  node "ip-192-168-68-54.us-west-2.compute.internal" is ready
[✔]  created 1 nodegroup(s) in cluster "demo-windows-cluster"
[ℹ]  checking security group configuration for all nodegroups
[ℹ]  all nodegroups have up-to-date configuration

**kubectl get secret -nkube-system vpc-admission-webhook-certs**
NAME                          TYPE     DATA   AGE
vpc-admission-webhook-certs   Opaque   2      16m

**kubectl get pods --all-namespaces**
NAMESPACE     NAME                                       READY   STATUS    RESTARTS   AGE
kube-system   aws-node-79mqm                             1/1     Running   0          28m
kube-system   aws-node-8gwmr                             1/1     Running   0          28m
kube-system   coredns-6f647f5754-5fdhq                   1/1     Running   0          33m
kube-system   coredns-6f647f5754-6whhk                   1/1     Running   0          33m
kube-system   kube-proxy-g8drl                           1/1     Running   0          28m
kube-system   kube-proxy-r52kq                           1/1     Running   0          28m
**kube-system   vpc-admission-webhook-6d4f7dc54f-nlnqt     1/1     Running   0          26m**
kube-system   vpc-resource-controller-5ddd7b84dd-bgslq   1/1     Running   0          26m
